internal/tasks/fleet_rollout.go

```diff
@@ -6,23 +6,46 @@ import (
 
 	api "github.com/flightctl/flightctl/api/v1alpha1"
 	"github.com/flightctl/flightctl/internal/store"
+	"github.com/flightctl/flightctl/internal/store/model"
+	"github.com/flightctl/flightctl/internal/util"
 	"github.com/flightctl/flightctl/pkg/log"
 	"github.com/flightctl/flightctl/pkg/reqid"
 	"github.com/go-chi/chi/v5/middleware"
-	"github.com/google/uuid"
 	"github.com/sirupsen/logrus"
 )
 
-/* Wait to be notified via channel about fleet template updates, exit upon ctx.Done() */
+// Wait to be notified via channel about fleet template updates, exit upon ctx.Done()
 func FleetRollouts(taskManager TaskManager) {
-	reqid.OverridePrefix("fleet-rollout")
-
 	for {
 		select {
-		case fleetRef := <-taskManager.channels[FleetTemplateRollout]:
-			err := rolloutFleet(taskManager.log, taskManager.store.Fleet(), taskManager.store.Device(), fleetRef)
-			if err != nil {
-				taskManager.log.Errorf("failed rolling out fleet %s/%s: %v", fleetRef.OrgID, fleetRef.Name, err)
+		case resourceRef := <-taskManager.channels[ChannelFleetTemplateRollout]:
+			requestID := reqid.NextRequestID()
+			ctx := context.WithValue(context.Background(), middleware.RequestIDKey, requestID)
+			log := log.WithReqIDFromCtx(ctx, taskManager.log)
+			logic := FleetRolloutsLogic{
+				ctx:         ctx,
```

Comment:
its generally considered an antipattern to attach ctx to a struct can we look at ways to pass it as a param?